### DISCUSSION TITLE
build and deploy 에러에서 npm run build를 찾지 못하는 문제

### DISCUSSION THREAD

**Discussion by Yenowme** [devrel-kr/HackaLearn#59] [2 Q&A - 클라우드 스킬 챌린지] [answered]
Azure Static Web Apps를 통한 사용자 인증 과제를 진행중에, 깃헙 액션을 통한 build and Deploy job이 #43 과 같은 오류를 띄워서 해당 질문의 해결방법대로 yml파일을 아래와 같이 수정했습니다. 

<img width="1069" alt="스크린샷 2021-08-05 오후 5 50 14" src="https://user-images.githubusercontent.com/76904042/128321403-c2131ae9-8464-42a1-a0a9-030aee201da7.png">

<img width="781" alt="스크린샷 2021-08-05 오후 4 36 06" src="https://user-images.githubusercontent.com/76904042/128310591-fa84852d-394b-40ec-a580-eb286a16a3de.png">
그런데 여기서는 동일한 오류가 해결되지 않고 위와 같이 뜹니다.

vue-app/package.json파일에 build스크립트가 잘 있고,
<img width="385" alt="스크린샷 2021-08-05 오후 5 27 10" src="https://user-images.githubusercontent.com/76904042/128318018-ba7eb27d-1d53-46ae-b683-1de4ebd7450e.png">

 디렉토리의 문제인가 해서 스크립트에 pwd 를 추가해서 돌려보니 
<img width="717" alt="스크린샷 2021-08-05 오후 5 26 03" src="https://user-images.githubusercontent.com/76904042/128317884-48d95cf0-9fdd-4072-967f-3d0b315a1bc3.png">
vue-app으로 맞게 떴는데 왜 저런 오류가 뜨는지 감이 안잡히네요.. 애저 클라우드에 직접 접속해서 알아보고싶기도 한데 방법을 모르겠고..
로컬에서는 build명령어가 잘 됩니다.

그 전 과제에서는 문제없이 해결되었는데, 왜 여기서는 오류가 해결되지 않는지 모르겠네요.. 아시는 분 있으실까요?

**Comment by kooljay82** [comment, marked as answer]
https://github.com/Yenowme/my-static-web-app-authn/runs/3250445388?check_suite_focus=true

이게 연결된 액션인가요??

**Reply by Yenowme** [reply]
네 맞습니다ㅜㅜ

**Reply by kooljay82** [reply]
샌드박스도 만료되지 않았고, 구독 구성 시, vue-app 디렉토리와 dist 설정이 제대로 되어있다면, 해당 문제를 해결하기 위해 조금 더 시간을 내어 문제를 면밀하게 봐야할 것 같습니다.

**Reply by kooljay82** [reply]
@Yenowme 님 지금 시간 있으세요??

*내용추가* 다른 용무 중이신 것 같아서 제가 해결한 방법 공유 드립니다.

`package-lock.json`을 삭제하시고 `vue-app` 디렉토리의 `.gitignore` 파일에 `package-lock.json`을 명시하신 후 변경사항 커밋과 푸시 해주세요. 

만약 위의 방법대로 진행했음에도 불구하고 오류가 발생한다면, yml에 삽입하셨던 스크립트(npm 버전을 올리는 부분)를 삭제해주시고 실행해주세요. 

로컬 머신 환경 정보

- node v14.17.4
- npm 7.20.3

CI/CD 환경 정보

- node v14.17.4
- npm 6.14.14

두 환경에서 모두 빌드 성공하였습니다.

[참고](https://github.com/kooljay82/my-static-web-app-authn/runs/3252191879?check_suite_focus=true)

**Reply by Yenowme** [reply]
답변주신 방법으로 드디어 해결했네요. 같이 고민해주셔서 너무 감사합니다! 👍👍

**Comment by justinyoo** [comment]
위에 언급한 에러가 발생한다면

환경 변수 값인 `CUSTOM_BUILD_COMMAND: "npm install -g npm && npm install --unsafe-perm && npm run build"` 부분을 추가하지 말고, 위의 @kooljay82 님께서 말씀하신대로 `package-lock.json` 파일을 빼고 푸시해 보세요. 그러면 성공합니다.

제 로그를 참고해 보세요: [바로가기](https://github.com/justinyoo/my-static-web-app-authn/runs/3252711914?check_suite_focus=true)

**Reply by justinyoo** [reply]
일단 관련 이슈를 남겨뒀습니다. https://github.com/Azure/static-web-apps/issues/534

**Reply by minhoryang** [reply]
Package.json에 build가 존재하는데 Npm run build가 실행되지 않으면 실행 위치를 생각해봐야합니다. App폴더 안에서 명령이 실행되도록 해야합니다

**Reply by justinyoo** [reply]
@minhoryang 오 맞습니다! 일반적인 상황이라면 그걸 먼저 의심해 봐야 하는데, 여기서는 깃헙 액션과 관련한 이슈로 보여욥.

**Reply by Yenowme** [reply]
답변 감사합니다! 파일 삭제하고 푸시하니 해결되네요 ..! 왜 run build가 안되는지 이유가 궁금하네요..흠..
